doc/source/user_guide/style.ipynb

```diff
@@ -335,28 +498,64 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "s = df.style.applymap(color_negative_red)\n",
-    "s"
+    "def highlight_max(s, props=''):\n",
+    "    return np.where(s == np.nanmax(s.values), props, '')\n",
+    "s2.apply(highlight_max, props='color:white;background-color:darkblue', axis=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx": "hidden"
+   },
+   "outputs": [],
+   "source": [
+    "# Hidden cell to avoid CSS clashes and latter code upcoding previous formatting \n",
+    "s2.set_uuid('after_apply')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the same function across the different axes, highlighting here the DataFrame maximum in purple, and row maximums in pink."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s2.apply(highlight_max, props='color:white;background-color:pink;', axis=1)\\\n",
+    "  .apply(highlight_max, props='color:white;background-color:purple', axis=None)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Notice the similarity with the standard `df.applymap`, which operates on DataFrames elementwise. We want you to be able to reuse your existing knowledge of how to interact with DataFrames.\n",
+    "This last example shows how some styles have been overwritten by others. In general the most recent style applied is effective but you can read more in the [section on CSS hierarchies](#CSS-Hierarchies). You can also apply these styles to more granular parts of the DataFrame - read more in section on [subset slicing](#Finer-Control-with-Slicing).\n",
+    "\n",
+    "It is possible to replicate some of this functionality using just classes but it can be more cumbersome. See [iii) as part of Optimization](#Optimization)\n",
     "\n",
-    "Notice also that our function returned a string containing the CSS attribute and value, separated by a colon just like in a `<style>` tag. This will be a common theme.\n",
+    "<div class=\"alert alert-info\">\n",
     "\n",
-    "Finally, the input shapes matched. `Styler.applymap` calls the function on each scalar input, and the function returns a scalar output."
+    "*Debugging Tip*: If you're having trouble writing your style function, try just passing it into ``DataFrame.apply``. Internally, ``Styler.apply`` uses ``DataFrame.apply`` so the result should be the same.\n",
```

Comment:
No I don't think so, seems pretty well tested. Also from my perspective this was a legacy cell I just kept included since it seemed helpful info.

Reply:
Then I think I don't understand the tip. It sounds like "if you're having trouble, then try this other way which should give you the same trouble." My original thought was it was saying "try DataFrame.apply instead in case we have a bug in Styler.apply"

Reply:
I think with DataFrame.apply you get to see the results as an output, whereas with Styler.apply it all gets baked in and the only results you see are either a HTML rendered tables (presumably without your intended styles) or some HTML code that is not necessarily structured in a 2D grid - so I feel the debugging tip was intended as an intermediate step.

Reply:
Thanks - this is what I was missing. Maybe add something like "this way it is more convenient to inspect the style being applied"?

Reply:
changed..

*Debugging Tip*: If you're having trouble writing your style function, try just passing it into ``DataFrame.apply``. Internally, ``Styler.apply`` uses ``DataFrame.apply`` so the result should be the same, and with ``DataFrame.apply`` you will be able to see the CSS string output of your intended function in each cell.